diff --git a/toolkit/content/aboutUrlClassifier.xhtml b/toolkit/content/aboutUrlClassifier.xhtml
index 333ef429edf87bd67c08d740d563ce5ca8d44af9..c26c2572cbb824f9a19c95475f7f07dbbeb23dd2 100644
--- a/toolkit/content/aboutUrlClassifier.xhtml
+++ b/toolkit/content/aboutUrlClassifier.xhtml
@@ -16,6 +16,7 @@
   <link rel="stylesheet" href="chrome://global/content/aboutUrlClassifier.css" type="text/css"/>
   <link rel="localization" href="toolkit/about/url-classifier.ftl"/>
   <script src="chrome://global/content/aboutUrlClassifier.js"></script>
+  <script src="chrome://browser/content/contentTheme.js"></script>
 </head>
 
 <body class="wide-container">
