diff --git a/toolkit/content/aboutProfiles.xhtml b/toolkit/content/aboutProfiles.xhtml
index d7079a022d0f5a8d44381580bd54cc2f99df53d2..1e9188a382b6676243c479fb2544fcea8387d800 100644
--- a/toolkit/content/aboutProfiles.xhtml
+++ b/toolkit/content/aboutProfiles.xhtml
@@ -16,6 +16,7 @@
     <script src="chrome://global/content/aboutProfiles.js" />
     <link rel="localization" href="branding/brand.ftl" />
     <link rel="localization" href="toolkit/about/aboutProfiles.ftl" />
+    <script src="chrome://browser/content/contentTheme.js"></script>
 </head>
   <body id="body" class="wide-container">
     <h1 data-l10n-id="profiles-title"></h1>
