diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index 55caf7b83f44330c79436bca4614ebbc18f0d4f3..5bac0976a9453a94ea0b194f550dd9e292c8e3dd 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -27,6 +27,7 @@
 #ifndef ANDROID
     <link rel="localization" href="toolkit/featuregates/features.ftl"/>
 #endif
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
 
   <body class="wide-container">
