diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index 4239a2af4952c0c7ceafaae1d614925a960402bc..f53ab91867941b7282d71b4687ae8af7df47e621 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -58,6 +58,7 @@ browser.jar:
         content/browser/browser-toolbarKeyNav.js            (content/browser-toolbarKeyNav.js)
         content/browser/browser-thumbnails.js               (content/browser-thumbnails.js)
         content/browser/browser-graphics-utils.js           (content/browser-graphics-utils.js)
+        content/browser/browser-verticaltabs.js             (content/browser-verticaltabs.js)
         content/browser/browser-webrtc.js                   (content/browser-webrtc.js)
 *       content/browser/pageinfo/pageInfo.xhtml             (content/pageinfo/pageInfo.xhtml)
         content/browser/pageinfo/pageInfo.js                (content/pageinfo/pageInfo.js)
