diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index fdf515554d67c5082b0d9c73b72533e06a085030..2d8130d90a6ff43af287debca7c44cc13d4da3f6 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -122,6 +122,22 @@
   <html:h1 data-l10n-id="language-and-appearance-header"/>
 </hbox>
 
+<!-- Sidebar -->
+<groupbox id="sidebarGroup" data-category="paneGeneral" hidden="true">
+  <label><html:h2 data-l10n-id="preferences-sidebar-header"/></label>
+
+  <checkbox id="sidebarVisible"
+    preference="pulse.sidebar.enabled"
+    data-l10n-id="preferences-sidebar-visible"/>
+  <checkbox id="sidebarKeepTabsActive"
+    preference="pulse.sidebar.keeptabsactive.enabled"
+    data-l10n-id="preferences-sidebar-active-tabs"/>
+  <description id="sidebarKeepTabsActiveDescription"
+    data-l10n-id="preferences-sidebar-active-tabs-description">
+    <html:a id="restart-browser-button" class="text-link" data-l10n-name="restart-browser" href="about:restartrequired"/>
+  </description>
+</groupbox>
+
 <!-- Website appearance -->
 <groupbox id="webAppearanceGroup" data-category="paneGeneral" hidden="true">
   <html:h2 data-l10n-id="preferences-web-appearance-header"/>
