diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index a4a4ec894115afc70d203661f2b225adeeeaa8fc..6e39c59690e115ea484f55a70eb964b114c087fc 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -161,6 +161,13 @@ Preferences.addAll([
 
   // Media
   { id: "media.hardwaremediakeys.enabled", type: "bool" },
+
+
+  // Pulse browser: custom settings
+  { id: 'pulse.sidebar.enabled', type: 'bool' },
+  { id: 'pulse.sidebar.extensions.enabled', type: 'bool' },
+  { id: 'pulse.sidebar.keeptabsactive.enabled', type: 'bool' },
+  { id: 'pulse.tabs.vertical', type: 'bool' }
 ]);
 
 if (AppConstants.HAVE_SHELL_SERVICE) {
