diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index a9a1511a8da36140be93264effe683e5d5007f11..1dbcad427fc40cd30a31e9bae9684977f7a74be5 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -61,6 +61,17 @@
           hidden="true">
     <label><html:h2 data-l10n-id="tabs-group-header"/></label>
 
+    <checkbox id="verticalTabsGroup"
+              preference="pulse.tabs.vertical"
+              data-l10n-id="preferences-vertical-tabs-enabled"/>
+
+    <vbox class="indent">
+        <checkbox id="verticalTabsCollapse"
+                  preference="pulse.tabs.vertical.collapse" 
+									data-l10n-id="preferences-vertical-tabs-collapse" />
+    </vbox>
+ 
+
     <checkbox id="ctrlTabRecentlyUsedOrder" data-l10n-id="ctrl-tab-recently-used-order"
               preference="browser.ctrlTab.sortByRecentlyUsed"/>
 
@@ -122,6 +133,15 @@
   <html:h1 data-l10n-id="language-and-appearance-header"/>
 </hbox>
 
+<!-- Sidebar -->
+<groupbox id="sidebarGroup" data-category="paneGeneral" hidden="true">
+  <label><html:h2 data-l10n-id="preferences-sidebar-header"/></label>
+
+  <checkbox id="sidebarVisible"
+    preference="pulse.sidebar.enabled"
+    data-l10n-id="preferences-sidebar-visible"/>
+</groupbox>
+
 <!-- Website appearance -->
 <groupbox id="webAppearanceGroup" data-category="paneGeneral" hidden="true">
   <html:h2 data-l10n-id="preferences-web-appearance-header"/>
