diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index a4a4ec894115afc70d203661f2b225adeeeaa8fc..304b3303c848fbecbfc20f134819c33b65a387af 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -161,6 +161,11 @@ Preferences.addAll([
 
   // Media
   { id: "media.hardwaremediakeys.enabled", type: "bool" },
+
+
+  // Pulse browser: custom settings
+  { id: 'pulse.sidebar.enabled', type: 'bool' },
+  { id: 'pulse.tabs.vertical', type: 'bool' }
 ]);
 
 if (AppConstants.HAVE_SHELL_SERVICE) {
L_SERVICE) {
