diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index 4cdcece2776506eb772a5463e245218837abc5e0..9b517ea6a3fc2ff43c7c412b90f241e75a65bdae 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -125,6 +125,22 @@
   <html:h1 data-l10n-id="language-and-appearance-header"/>
 </hbox>
 
+<!-- Sidebar -->
+<groupbox id="sidebarGroup" data-category="paneGeneral" hidden="true">
+  <label><html:h2 data-l10n-id="preferences-sidebar-header"/></label>
+
+  <checkbox id="sidebarVisible"
+    preference="pulse.sidebar.enabled"
+    data-l10n-id="preferences-sidebar-visible"/>
+  <checkbox id="sidebarKeepTabsActive"
+    preference="pulse.sidebar.keeptabsactive.enabled"
+    data-l10n-id="preferences-sidebar-active-tabs"/>
+  <description id="sidebarKeepTabsActiveTooltip"
+    data-l10n-id="preferences-sidebar-active-tabs-tooltip">
+    <html:a id="restart-browser-button" class="text-link" data-l10n-name="restart-browser" href="about:restartrequired"/>
+  </description>
+</groupbox>
+
 <!-- Website appearance -->
 <groupbox id="webAppearanceGroup" data-category="paneGeneral" hidden="true">
   <html:h2 data-l10n-id="preferences-web-appearance-header"/>
