diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index ceee50df3743724f1d59cebb0fec244639e65405..8e065ec0aec2ec86ad2d51ccd828afd36cd0e212 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -154,6 +154,11 @@ Preferences.addAll([
 
   // Media
   { id: "media.hardwaremediakeys.enabled", type: "bool" },
+
+
+  // Pulse browser: custom settings
+  { id: 'pulse.sidebar.enabled', type: 'bool' },
+  { id: 'pulse.sidebar.extensions.enabled', type: 'bool' }
 ]);
 
 if (AppConstants.HAVE_SHELL_SERVICE) {
