diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 0dc01225f51934fd86653748c1182960b5ba4e70..3f24c789c550021cd9be90691aac167686f550da 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -161,6 +161,12 @@ Preferences.addAll([
 
   // Media
   { id: "media.hardwaremediakeys.enabled", type: "bool" },
+
+
+  // Pulse browser: custom settings
+  { id: 'pulse.sidebar.enabled', type: 'bool' },
+  { id: 'pulse.sidebar.extensions.enabled', type: 'bool' },
+  { id: 'pulse.sidebar.keeptabsactive.enabled', type: 'bool' }
 ]);
 
 if (AppConstants.HAVE_SHELL_SERVICE) {
