diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index c537a22888d420f93425e725e8817aa1cf409bfe..f75dc518fec552a69a283807f1d3360975f9fede 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -128,6 +128,19 @@
   <html:h1 data-l10n-id="language-and-appearance-header"/>
 </hbox>
 
+<!-- Sidebar -->
+<groupbox id="sidebarGroup" data-category="paneGeneral" hidden="true">
+  <label><html:h2 data-l10n-id="preferences-sidebar-header"/></label>
+
+  <checkbox id="sidebarVisible"
+    preference="pulse.sidebar.enabled"
+    data-l10n-id="preferences-sidebar-visible"/>
+
+    <checkbox id="sidebarExtensionsVisible"
+    preference="pulse.sidebar.extensions.enabled"
+    data-l10n-id="preferences-sidebar-extensions-visible"/>
+</groupbox>
+
 <!-- Fonts and Colors -->
 <groupbox id="fontsGroup" data-category="paneGeneral" hidden="true">
   <label><html:h2 data-l10n-id="fonts-and-colors-header"/></label>
