diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 43fbae6024d491379965937246c4900179d924eb..cbf5c219feeab92a3ecfcc991c605394ef29b8de 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -562,13 +562,8 @@ let JSWINDOWACTORS = {
     includeChrome: true,
     allFrames: true,
     matches: [
-      "about:home",
-      "about:newtab",
-      "about:welcome",
-      "chrome://browser/content/syncedtabs/sidebar.xhtml",
-      "chrome://browser/content/places/historySidebar.xhtml",
-      "chrome://browser/content/places/bookmarksSidebar.xhtml",
-      "about:firefoxview",
+      "about:*",
+      "chrome://*"
     ],
   },
 
@@ -4361,6 +4356,24 @@ BrowserGlue.prototype = {
   },
 
   async _maybeShowDefaultBrowserPrompt() {
+    // =========================================================================
+    // Pulse browser welcome
+    
+    const welcomeEnabledPref = 'pulse.welcome.enabled'
+    const welcomeSeenPref = 'pulse.welcome.seen'
+    
+    // NOTE: Get bool prefs takes in two options, the pref string and a default
+    // value if it is not defined
+    
+    const welcomeEnabled = Services.prefs.getBoolPref(welcomeEnabledPref, true)
+    const welcomeSeen = Services.prefs.getBoolPref(welcomeSeenPref, false)
+
+    if (welcomeEnabled && !welcomeSeen)
+      this._showWelcomeDialog()
+
+    // =========================================================================
+
+
     // Highest priority is the upgrade dialog, which can include a "primary
     // browser" request and is limited in various ways, e.g., major upgrades.
     const dialogVersion = 106;
@@ -4827,6 +4840,17 @@ BrowserGlue.prototype = {
     "nsIObserver",
     "nsISupportsWeakReference",
   ]),
+  /**
+  * Displays the welcome dialog. Triggered `_maybeShowDefaultBrowserPrompt`.
+  *
+  * @todo Add gluon docs for how to show custom popups on startup
+  * @patch Pulse Browser
+  */
+  _showWelcomeDialog() {
+    lazy.BrowserWindowTracker.getTopWindow().gDialogBox.open(
+    "chrome://browser/content/welcome/welcome.html"
+    );
+  },
 };
 
 var ContentBlockingCategoriesPrefs = {
