diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 64ac529d54270758b6dfa7a0a04b604b89a92402..59845d228da66c087e59f09052b1e482ddbac6cb 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -4373,6 +4373,23 @@ BrowserGlue.prototype = {
       return;
     }
 
+    // =========================================================================
+    // Pulse browser welcome
+
+    const welcomeEnabledPref = 'pulse.welcome.enabled'
+    const welcomeSeenPref = 'pulse.welcome.seen'
+
+    // NOTE: Get bool prefs takes in two options, the pref string and a default
+    // value if it is not defined
+
+    const welcomeEnabled = Services.prefs.getBoolPref(welcomeEnabledPref, true)
+    const welcomeSeen = Services.prefs.getBoolPref(welcomeSeenPref, false)
+
+    if (welcomeEnabled && !welcomeSeen)
+      this._showWelcomeDialog()
+
+    // =========================================================================
+
     const willPrompt = await DefaultBrowserCheck.willCheckDefaultBrowser(
       /* isStartupCheck */ true
     );
@@ -4780,6 +4797,18 @@ BrowserGlue.prototype = {
     "nsIObserver",
     "nsISupportsWeakReference",
   ]),
+
+  /**
+   * Displays the welcome dialog. Trigured `_maybeShowDefaultBrowserPrompt`.
+   *
+   * @todo Add gluon docs for how to show custom popups on startup
+   * @patch Pulse Browser
+   */
+  _showWelcomeDialog() {
+    BrowserWindowTracker.getTopWindow().gDialogBox.open(
+      "chrome://browser/content/welcome/welcome.html"
+    );
+  },
 };
 
 var ContentBlockingCategoriesPrefs = {
