diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 0c16eba15229d0003d62bb76307affe944d7c47b..c37ab06ae9bd6b4b00e97b35563456e9cf7b5cb7 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -4045,6 +4045,18 @@ BrowserGlue.prototype = {
     );
   },
 
+  /**
+   * Displays the welcome dialog. Trigured `_maybeShowDefaultBrowserPrompt`.
+   *
+   * @todo Provide documentation for implementing custom popups
+   * @patch FocusBrowser
+   */
+  _showWelcomeDialog() {
+    BrowserWindowTracker.getTopWindow().gDialogBox.open(
+      "chrome://browser/content/welcome/welcome.html"
+    );
+  },
+
   async _maybeShowDefaultBrowserPrompt() {
     // Highest priority is the upgrade dialog, which can include a "primary
     // browser" request and is limited in various ways, e.g., major upgrades.
@@ -4102,6 +4114,25 @@ BrowserGlue.prototype = {
       return;
     }
 
+
+    // =========================================================================
+    // Focus browser welcome
+
+    const welcomeEnabledPref = 'focus.welcome.enabled'
+    const welcomeSeenPref = 'focus.welcome.seen'
+
+    // NOTE: Get bool prefs takes in two options, the pref string and a default
+    // value if it is not defined
+
+    const welcomeEnabled = Services.prefs.getBoolPref(welcomeEnabledPref, true)
+    const welcomeSeen = Services.prefs.getBoolPref(welcomeSeenPref, false)
+
+    if (welcomeEnabled && !welcomeSeen)
+      this._showWelcomeDialog()
+
+    // =========================================================================
+
+
     const willPrompt = await DefaultBrowserCheck.willCheckDefaultBrowser(
       /* isStartupCheck */ true
     );
