diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 89651a029abc4f41205f099950a0a21eddb26cc1..9b793db6fc17848670cf18ca1d9552960a57f0ea 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -4183,6 +4183,18 @@ BrowserGlue.prototype = {
     SpecialMessageActions.handleAction(config, browser);
   },
 
+  /**
+   * Displays the welcome dialog. Trigured `_maybeShowDefaultBrowserPrompt`.
+   *
+   * @todo Provide documentation for implementing custom popups
+   * @patch FocusBrowser
+   */
+  _showWelcomeDialog() {
+    BrowserWindowTracker.getTopWindow().gDialogBox.open(
+      "chrome://browser/content/welcome/welcome.html"
+    );
+  },
+
   async _maybeShowDefaultBrowserPrompt() {
     // Highest priority is the upgrade dialog, which can include a "primary
     // browser" request and is limited in various ways, e.g., major upgrades.
@@ -4236,6 +4248,23 @@ BrowserGlue.prototype = {
       return;
     }
 
+    // =========================================================================
+    // Pulse browser welcome
+
+    const welcomeEnabledPref = 'pulse.welcome.enabled'
+    const welcomeSeenPref = 'pulse.welcome.seen'
+
+    // NOTE: Get bool prefs takes in two options, the pref string and a default
+    // value if it is not defined
+
+    const welcomeEnabled = Services.prefs.getBoolPref(welcomeEnabledPref, true)
+    const welcomeSeen = Services.prefs.getBoolPref(welcomeSeenPref, false)
+
+    if (welcomeEnabled && !welcomeSeen)
+      this._showWelcomeDialog()
+
+    // =========================================================================
+
     const willPrompt = await DefaultBrowserCheck.willCheckDefaultBrowser(
       /* isStartupCheck */ true
     );
