diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index cee46e6341f98441df743f4f5abccc380a67f74b..728e4e6d4b708d75a266181052e4af54c17b6180 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -4343,6 +4343,24 @@ BrowserGlue.prototype = {
   },
 
   async _maybeShowDefaultBrowserPrompt() {
+    // =========================================================================
+    // Pulse browser welcome
+    
+    const welcomeEnabledPref = 'pulse.welcome.enabled'
+    const welcomeSeenPref = 'pulse.welcome.seen'
+    
+    // NOTE: Get bool prefs takes in two options, the pref string and a default
+    // value if it is not defined
+    
+    const welcomeEnabled = Services.prefs.getBoolPref(welcomeEnabledPref, true)
+    const welcomeSeen = Services.prefs.getBoolPref(welcomeSeenPref, false)
+
+    if (welcomeEnabled && !welcomeSeen)
+      this._showWelcomeDialog()
+
+    // =========================================================================
+
+
     // Highest priority is the upgrade dialog, which can include a "primary
     // browser" request and is limited in various ways, e.g., major upgrades.
     const dialogVersion = 106;
@@ -4809,6 +4827,17 @@ BrowserGlue.prototype = {
     "nsIObserver",
     "nsISupportsWeakReference",
   ]),
+  /**
+  * Displays the welcome dialog. Trigured `_maybeShowDefaultBrowserPrompt`.
+  *
+  * @todo Add gluon docs for how to show custom popups on startup
+  * @patch Pulse Browser
+  */
+  _showWelcomeDialog() {
+    lazy.BrowserWindowTracker.getTopWindow().gDialogBox.open(
+    "chrome://browser/content/welcome/welcome.html"
+    );
+  },
 };
 
 var ContentBlockingCategoriesPrefs = {
