diff --git a/browser/components/moz.build b/browser/components/moz.build
index 945a55a1b216dac9f5b66d883bf94712a5994311..ae6853e34165a6b8fb4556c9fbfd701dccff1b50 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -25,7 +25,6 @@ with Files("safebrowsing/**"):
 with Files("controlcenter/**"):
     BUG_COMPONENT = ("Firefox", "General")
 
-
 DIRS += [
     "about",
     "aboutlogins",
@@ -60,6 +59,9 @@ DIRS += [
     "urlbar",
 ]
 
+# Pulse browser imports
+DIRS += ["pulseTheme", "welcome"]
+
 DIRS += ["build"]
 
 if CONFIG["NIGHTLY_BUILD"]:
