diff --git a/browser/components/moz.build b/browser/components/moz.build
index 5a45200d348199e6bf0aa1e49c37a7d6129a376a..c7bf38f4341f939e38b23de7eb814a794af4a798 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -60,6 +60,9 @@ DIRS += [
     "urlbar",
 ]
 
+# Pulse: Custom components
+DIRS += ["welcome"]
+
 DIRS += ["build"]
 
 if CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
