diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 9b56ba74cff12f029b6d8910ff487a216a15194f..ec90cd53380cd948d4aaee457c257fdeea7023ea 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -2781,3 +2781,5 @@ pref("browser.pdfjs.feature-tour", "{\"screen\":\"\",\"complete\":false}");
 
 // Enables the cookie banner desktop UI.
 pref("cookiebanners.ui.desktop.enabled", false);
+
+#include pulse-browser.js
