diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 81f4d49068c285049d49db6c8fc5c0fbe042ff6b..39ef13fb66567c31ebf932d7a6f528fa1d635069 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -2804,3 +2804,5 @@ pref("cookiebanners.ui.desktop.cfrVariant", 0);
   pref("browser.swipe.navigation-icon-min-radius", 12);
   pref("browser.swipe.navigation-icon-max-radius", 20);
 #endif
+
+#include pulse-browser.js
