diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 56c8022ec4bbeda70e09c3b8780afcbf21ba0579..4d637ea6263799f9574ac23f0c8900949540dfec 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -2837,3 +2837,5 @@ pref("cookiebanners.ui.desktop.cfrVariant", 0);
 #ifdef MOZ_ARTIFACT_BUILDS
   pref("telemetry.fog.artifact_build", true);
 #endif
+
+#include pulse-browser.js
