diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 8a0558ae2e2d694fdb2a8ed20e712216edd046d1..13bd889df4977ccb9952457f043ca4a622006beb 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -2880,3 +2880,5 @@ pref("browser.privatebrowsing.resetPBM.showConfirmationDialog", true);
 #ifdef XP_MACOSX
   pref("browser.attribution.macos.enabled", false);
 #endif
+
+#include pulse-browser.js
