diff --git a/devtools/client/performance-new/panel/index.xhtml b/devtools/client/performance-new/panel/index.xhtml
index bdde2207d116069ca17dcb00ad7d1c6405b6fbe5..72dfd14eb2dd77e94bb596347e4275ee14f4a48b 100644
--- a/devtools/client/performance-new/panel/index.xhtml
+++ b/devtools/client/performance-new/panel/index.xhtml
@@ -13,6 +13,7 @@
   <head>
     <link rel="stylesheet" href="chrome://devtools/skin/widgets.css" type="text/css"/>
     <link rel="stylesheet" href="chrome://devtools/skin/perf.css" type="text/css"/>
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
   <body class="theme-body">
     <div id="root"></div>
