diff --git a/devtools/client/performance-new/aboutprofiling/index.xhtml b/devtools/client/performance-new/aboutprofiling/index.xhtml
index 010a7776d0ce47708d5bcb3c4051edcc6f34e279..2b36f65a52ddaf1ff3b81d765e0eabd94913c278 100644
--- a/devtools/client/performance-new/aboutprofiling/index.xhtml
+++ b/devtools/client/performance-new/aboutprofiling/index.xhtml
@@ -10,6 +10,7 @@
     <link rel="stylesheet" href="chrome://global/skin/global.css" type="text/css"/>
     <link rel="stylesheet" href="chrome://global/skin/in-content/common.css" type="text/css"/>
     <link rel="stylesheet" href="chrome://devtools/skin/aboutprofiling.css" type="text/css"/>
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
   <body class="theme-body">
     <div id="root"></div>
