diff --git a/browser/base/content/aboutDialog.xhtml b/browser/base/content/aboutDialog.xhtml
index 90a568a17dd6f1690ccc55a2f927d5cfec6c722b..5fa2a994d1f4d664703e7c6c5830aa9618329ed0 100644
--- a/browser/base/content/aboutDialog.xhtml
+++ b/browser/base/content/aboutDialog.xhtml
@@ -41,6 +41,7 @@
     <hbox id="clientBox">
       <vbox id="leftBox" flex="1"/>
       <vbox id="rightBox" flex="1">
+        <label id="logo" data-l10n-id="aboutDialog-header"></label>
         <label id="release" hidden="true">
         <!-- This string is explicitly not translated -->
           Extended Support Release
@@ -141,12 +142,12 @@
             </description>
           </vbox>
           <description class="text-blurb" id="communityDesc" data-l10n-id="community-2">
-            <label is="text-link" href="https://www.mozilla.org/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="community-mozillaLink"/>
-            <label is="text-link" useoriginprincipal="true" href="about:credits" data-l10n-name="community-creditsLink"/>
+            <label is="text-link" href="https://fushra.com/" data-l10n-name="community-mozillaLink"/>
+            <label is="text-link" useoriginprincipal="true" href="https://github.com/pulse-browser/browser/graphs/contributors" data-l10n-name="community-creditsLink"/>
           </description>
           <description class="text-blurb" id="contributeDesc" data-l10n-id="helpus">
-            <label is="text-link" href="https://donate.mozilla.org/?utm_source=firefox&#38;utm_medium=referral&#38;utm_campaign=firefox_about&#38;utm_content=firefox_about" data-l10n-name="helpus-donateLink"/>
-            <label is="text-link" href="https://www.mozilla.org/contribute/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="helpus-getInvolvedLink"/>
+            <label is="text-link" href="https://github.com/pulse-browser/browser" data-l10n-name="helpus-donateLink"/>
+            <label is="text-link" href="https://github.com/pulse-browser/browser" data-l10n-name="helpus-getInvolvedLink"/>
           </description>
         </vbox>
       </vbox>
