diff --git a/browser/base/content/browser-sidebar.js b/browser/base/content/browser-sidebar.js
index eb1ee9e2b868fa06af4ed88e1d54583cd4196880..4a4f2e926d2954003cd4b8cbacc318dc820f8cc8 100644
--- a/browser/base/content/browser-sidebar.js
+++ b/browser/base/content/browser-sidebar.js
@@ -2,10 +2,14 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+
 /**
  * SidebarUI controls showing and hiding the browser sidebar.
  */
 var SidebarUI = {
+  /**
+   * @returns {Map<string, { title: string, url?: string, menuId: string, triggerButtonId?: string, iconurl: string, bottom?: string }>}
+   */
   get sidebars() {
     if (this._sidebars) {
       return this._sidebars;
@@ -19,6 +23,8 @@ var SidebarUI = {
             .getAttribute("label"),
           url: "chrome://browser/content/places/bookmarksSidebar.xhtml",
           menuId: "menu_bookmarksSidebar",
+          iconurl: "chrome://global/skin/icons/bookmark-outline.svg",
+          bottom: false,
         },
       ],
       [
@@ -30,6 +36,8 @@ var SidebarUI = {
           url: "chrome://browser/content/places/historySidebar.xhtml",
           menuId: "menu_historySidebar",
           triggerButtonId: "appMenuViewHistorySidebar",
+          iconurl: "chrome://browser/skin/history.svg",
+          bottom: false,
         },
       ],
       [
@@ -40,6 +48,8 @@ var SidebarUI = {
             .getAttribute("label"),
           url: "chrome://browser/content/syncedtabs/sidebar.xhtml",
           menuId: "menu_tabsSidebar",
+          iconurl: "chrome://browser/skin/tab.svg",
+          bottom: false,
         },
       ],
     ]));
@@ -53,7 +63,10 @@ var SidebarUI = {
     }
     return (this._browser = document.getElementById("sidebar"));
   },
+
   POSITION_START_PREF: "sidebar.position_start",
+  SIDEBAR_TABS_PREF: "pulse.sidebar.enabled",
+  SIDEBAR_EXTENSIONS_PREF: "pulse.sidebar.extensions.enabled",
   DEFAULT_SIDEBAR_ID: "viewBookmarksSidebar",
 
   // lastOpenedId is set in show() but unlike currentID it's not cleared out on hide
@@ -71,6 +84,8 @@ var SidebarUI = {
   },
   _splitter: null,
   _icon: null,
+  _sidebarIcons: null,
+  _sidebarBottomIcons: null,
   _reversePositionButton: null,
   _switcherPanel: null,
   _switcherTarget: null,
@@ -91,6 +106,10 @@ var SidebarUI = {
     this._box = document.getElementById("sidebar-box");
     this._splitter = document.getElementById("sidebar-splitter");
     this._icon = document.getElementById("sidebar-icon");
+
+    this._sidebarIcons = document.getElementById("sidebar-icons");
+    this._sidebarBottomIcons = document.getElementById("sidebar-icons-bottom");
+
     this._reversePositionButton = document.getElementById(
       "sidebar-reverse-position"
     );
@@ -102,9 +121,53 @@ var SidebarUI = {
       this.toggleSwitcherPanel();
     });
 
+    const sidebaritems = this.sidebars.keys();
+    const sidebarExtensionVis = Services.prefs.getBoolPref(this.SIDEBAR_EXTENSIONS_PREF);
+    for (const sidebaritem of sidebaritems) {
+      if(this.sidebars.get(sidebaritem).extensionId && !sidebarExtensionVis) {
+        return;
+      }
+      this.createSidebarItem(sidebaritem, this.sidebars.get(sidebaritem).iconurl, this.sidebars.get(sidebaritem));
+    }
+
+    const sidebarVisible = Services.prefs.getBoolPref(this.SIDEBAR_TABS_PREF, true)
+    this.setSidebarVisibility(sidebarVisible)
+
+    // Keep track on the changes of the sidebar visibility
+    Services.prefs.addObserver(this.SIDEBAR_TABS_PREF, this);
+
     this._inited = true;
 
     this._initDeferred.resolve();
+
+  },
+
+  /**
+   * Set the sidebar tabs to be visible or not visible
+   * @param {boolean} visible Sets the sidebar to be visible or not visible
+   */
+  setSidebarVisibility(visible) {
+    const sidebarContainer = document.getElementById('sidebar-container')
+
+    if (visible) {
+      sidebarContainer.style.display = 'flex'
+    } else {
+      sidebarContainer.style.display = 'none'
+    }
+  },
+
+  // nsIObserver
+
+  observe(aSubject, aTopic, aData) {
+    switch (aTopic) {
+      case "nsPref:changed":
+        switch (aData) {
+          case this.SIDEBAR_TABS_PREF:
+            this.setSidebarVisibility(Services.prefs.getBoolPref(this.SIDEBAR_TABS_PREF, true))
+            break
+        }
+        break;
+    }
   },
 
   uninit() {
@@ -135,6 +198,49 @@ var SidebarUI = {
     }
   },
 
+  createSidebarItem(id, icon, config) {
+    // if the sidebar item is already created, do nothing
+    if (document.getElementById(`sidebar-background-${id}`)) {
+      return;
+    }
+
+    const background = document.createElement("div")
+    background.classList.add("sidebar-item-background")
+    background.setAttribute("id", `sidebar-background-${id}`);
+    background.onclick = () => {
+      this.toggle(id)
+    }
+    
+    const iconEl = document.createElement("image");
+    iconEl.setAttribute("id", `sidebar-icon-${id}`);
+    iconEl.setAttribute("src", `${icon}`);
+    iconEl.style.backgroundImage = `url(${icon})`;
+    
+    background.append(iconEl)
+
+    if(config.bottom){
+      this._sidebarBottomIcons.appendChild(background);
+      return;
+    }
+    this._sidebarIcons.appendChild(background);
+    setTimeout(() => {
+      iconEl.setAttribute("class", "sidebar-icon-item");
+    }, 100);
+  },
+
+  deleteSidebarItem(id) {
+    const background = document.getElementById(`sidebar-background-${id}`)
+    const icon = document.getElementById(`sidebar-icon-${id}`)
+    if(background){
+      background.remove()
+    }
+    if(icon){
+      icon.remove()
+    }
+  },
+
+
+  
   /**
    * Opens the switcher panel if it's closed, or closes it if it's open.
    */
@@ -457,6 +563,10 @@ var SidebarUI = {
     return this._show(commandID).then(() => {
       this._loadSidebarExtension(commandID);
 
+      // Pulse: Reset sidebar margin to zero to allow for correct animations to 
+      // take place (adapted from dot browser, se below)
+      this._box.style.marginLeft = '0px'
+
       if (triggerNode) {
         updateToggleControlLabel(triggerNode);
       }
@@ -552,6 +662,25 @@ var SidebarUI = {
 
     this.selectMenuItem("");
 
+    // The below code was, in part, based on Dot Browser's source code
+    // https://github.com/dothq/browser-desktop/blob/3c6b992af8946e1c30180cc824a78d4bb85ba868/patches/browser-base-content-browser-sidebar-js.patch#L183
+
+    // Pulse Browser: Helper stuff for sidebar animation. We can fake sliding in
+    // and out by changing the left margin to be a negative value. Then, inside
+    // of our css code, we can animate it to close.
+    this._box.style.marginLeft = `${-Math.abs(this._box.getBoundingClientRect().width)}px`
+    
+    // We want to fully set it to hidden to slightly improve browser performance
+    // when it is hidden. Note that this replaces some of Mozilla's code that
+    // would generally be down below
+    setTimeout(() => {
+      this._box.hidden = true
+      this._splitter.hidden = true
+
+      this._box.removeAttribute("checked")
+    }, 170);
+
+
     // Replace the document currently displayed in the sidebar with about:blank
     // so that we can free memory by unloading the page. We need to explicitly
     // create a new content viewer because the old one doesn't get destroyed
@@ -560,9 +689,6 @@ var SidebarUI = {
     this.browser.setAttribute("src", "about:blank");
     this.browser.docShell.createAboutBlankContentViewer(null, null);
 
-    this._box.removeAttribute("checked");
-    this._box.hidden = this._splitter.hidden = true;
-
     let selBrowser = gBrowser.selectedBrowser;
     selBrowser.focus();
     if (triggerNode) {
@@ -575,21 +701,31 @@ var SidebarUI = {
    * none if the argument is an empty string.
    */
   selectMenuItem(commandID) {
+    // Pulse modification: This needs to update both which sidebar item is
+    // checked and which dropdown item is checked. Hence, we are overriding this
+    // method.
+    
     for (let [id, { menuId, triggerButtonId }] of this.sidebars) {
       let menu = document.getElementById(menuId);
-      let triggerbutton =
-        triggerButtonId && document.getElementById(triggerButtonId);
+      const dropdownTriggerButton = triggerButtonId && document.getElementById(triggerButtonId);
+      let sidebarTriggerButton = document.getElementById("sidebar-background-" + id);
+
+      setCheckedForItem(id, menu, sidebarTriggerButton);
+      setCheckedForItem(id, menu, dropdownTriggerButton);
+    }
+
+    function setCheckedForItem(id, menu, sidebarTriggureButton) {
       if (id == commandID) {
         menu.setAttribute("checked", "true");
-        if (triggerbutton) {
-          triggerbutton.setAttribute("checked", "true");
-          updateToggleControlLabel(triggerbutton);
+        if (sidebarTriggureButton) {
+          sidebarTriggureButton.setAttribute("checked", "true");
+          updateToggleControlLabel(sidebarTriggureButton);
         }
       } else {
         menu.removeAttribute("checked");
-        if (triggerbutton) {
-          triggerbutton.removeAttribute("checked");
-          updateToggleControlLabel(triggerbutton);
+        if (sidebarTriggureButton) {
+          sidebarTriggureButton.removeAttribute("checked");
+          updateToggleControlLabel(sidebarTriggureButton);
         }
       }
     }
