diff --git a/browser/base/content/browser-box.inc.xhtml b/browser/base/content/browser-box.inc.xhtml
index ae5b1979921a9390ee899d72b04bc4ff293b39d8..7a18f16b76a5a7fa9566e1f02fbae9d80e1212b5 100644
--- a/browser/base/content/browser-box.inc.xhtml
+++ b/browser/base/content/browser-box.inc.xhtml
@@ -3,9 +3,13 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 <hbox flex="1" id="browser">
+  <div id="sidebar-container" class="sidebar-container"> 
+    <vbox id="sidebar-icons"></vbox>
+    <vbox id="sidebar-icons-bottom"></vbox>
+  </div>
   <vbox id="sidebar-box" hidden="true" class="chromeclass-extrachrome">
     <box id="sidebar-header" align="center">
-      <toolbarbutton id="sidebar-switcher-target" class="tabbable">
+      <toolbarbutton id="sidebar-switcher-target" flex="1" class="tabbable">
         <image id="sidebar-icon" consumeanchor="sidebar-switcher-target"/>
         <label id="sidebar-title" crop="end" control="sidebar"/>
         <image id="sidebar-switcher-arrow"/>
