diff --git a/browser/base/content/browser-sidebar.js b/browser/base/content/browser-sidebar.js
index 22c886b09373a38a4fbceb6d279a58f276675d63..7b41613b2b5abe00e506a6b60e6bdf9a7670405e 100644
--- a/browser/base/content/browser-sidebar.js
+++ b/browser/base/content/browser-sidebar.js
@@ -11,6 +11,9 @@ var SidebarUI = {
       return this._sidebars;
     }
 
+    /**
+     * @returns {Map<string, { title: string, url?: string, onclick?: string, menuId?: string, triggerButtonId?: string, iconurl: string, isBottom: boolean }>}
+     */
     function makeSidebar({ elementId, ...rest }) {
       return {
         get sourceL10nEl() {
@@ -19,6 +22,7 @@ var SidebarUI = {
         get title() {
           return document.getElementById(elementId).getAttribute("label");
         },
+        isBottom: false,
         ...rest,
       };
     }
@@ -30,6 +34,7 @@ var SidebarUI = {
           elementId: "sidebar-switcher-bookmarks",
           url: "chrome://browser/content/places/bookmarksSidebar.xhtml",
           menuId: "menu_bookmarksSidebar",
+          iconurl: "chrome://global/skin/icons/bookmark-outline.svg",
         }),
       ],
       [
@@ -39,6 +44,7 @@ var SidebarUI = {
           url: "chrome://browser/content/places/historySidebar.xhtml",
           menuId: "menu_historySidebar",
           triggerButtonId: "appMenuViewHistorySidebar",
+          iconurl: "chrome://browser/skin/history.svg",
         }),
       ],
       [
@@ -47,8 +53,37 @@ var SidebarUI = {
           elementId: "sidebar-switcher-tabs",
           url: "chrome://browser/content/syncedtabs/sidebar.xhtml",
           menuId: "menu_tabsSidebar",
+          iconurl: "chrome://browser/skin/tab.svg",
         }),
       ],
+      [
+        "viewDownloadsSidebar",
+        makeSidebar({
+          elementId: "sidebar-switcher-downloads",
+          title: "Downloads",
+          url: "about:downloads",
+          menuId: "menu_downloadsSidebar",
+          iconurl: "chrome://browser/skin/downloads/downloads.svg",
+        }),
+      ],
+      [
+        "sidebar-view-addons",
+        {
+          title: "Add-ons",
+          onclick: "BrowserOpenAddonsMgr();",
+          iconurl: "chrome://mozapps/skin/extensions/category-themes.svg",
+          isBottom: true,
+        },
+      ],
+      [
+        "sidebar-view-preferances",
+        {
+          title: "Preferences",
+          onclick: "openPreferences();",
+          iconurl: "chrome://global/skin/icons/settings.svg",
+          isBottom: true,
+        },
+      ],
     ]));
   },
 
@@ -63,6 +98,10 @@ var SidebarUI = {
   POSITION_START_PREF: "sidebar.position_start",
   DEFAULT_SIDEBAR_ID: "viewBookmarksSidebar",
 
+  SIDEBAR_KEEP_TABS_ACTIVE_PREF: "pulse.sidebar.keeptabsactive.enabled",
+  SIDEBAR_TABS_PREF: "pulse.sidebar.enabled",
+  SIDEBAR_EXTENSIONS_PREF: "pulse.sidebar.extensions.enabled",
+
   // lastOpenedId is set in show() but unlike currentID it's not cleared out on hide
   // and isn't persisted across windows
   lastOpenedId: null,
@@ -83,6 +122,15 @@ var SidebarUI = {
   _switcherArrow: null,
   _inited: false,
 
+  _sidebarIcons: null,
+  _sidebarIconsBottom: null,
+  /** @type {HTMLDivElement?} */
+  _sidebarTabs: null,
+  /** @type {string[]} */
+  loadedSidebars: null,
+  /** @type {boolean} */
+  _keepTabsActivePref: false,
+
   /**
    * @type {MutationObserver | null}
    */
@@ -112,6 +160,51 @@ var SidebarUI = {
       this.toggleSwitcherPanel();
     });
 
+    this._sidebarIcons = document.getElementById("sidebar-icons");
+    this._sidebarIconsBottom = document.getElementById("sidebar-icons-bottom");
+    this._sidebarTabs = document.getElementById("sidebar-container");
+
+    const sidebaritems = this.sidebars.keys();
+    const sidebarExtensionVis = Services.prefs.getBoolPref(
+      this.SIDEBAR_EXTENSIONS_PREF,
+      false
+    );
+
+    this._keepTabsActivePref = Services.prefs.getBoolPref(
+      this.SIDEBAR_KEEP_TABS_ACTIVE_PREF,
+      false
+    );
+
+    for (const sidebaritem of sidebaritems) {
+      if (this.sidebars.get(sidebaritem).extensionId && !sidebarExtensionVis) {
+        return;
+      }
+      this.createSidebarItem(sidebaritem, this.sidebars.get(sidebaritem), true);
+    }
+
+    let sidebarVisible = false;
+    if (window.toolbar.visible) {
+      sidebarVisible = Services.prefs.getBoolPref(this.SIDEBAR_TABS_PREF, true);
+    }
+
+    this.setSidebarVisibility(sidebarVisible);
+
+    // Keep track on the changes of the sidebar visibility
+    Services.prefs.addObserver(this.SIDEBAR_TABS_PREF, this);
+
+    Services.prefs.addObserver(this.SIDEBAR_KEEP_TABS_ACTIVE_PREF, this);
+
+    //Check the pre-existance of sidebar and if it is not there, create it
+    var sidebar = this.browser;
+    let sidebarExists = !!sidebar;
+    if (this._keepTabsActivePref && sidebarExists) {
+      sidebar.remove();
+    } else if (this._keepTabsActivePref && !sidebarExists) {
+      this.createBrowser("sidebar");
+    }
+
+    this.loadedSidebars = [];
+
     this._inited = true;
 
     Services.obs.addObserver(this, "intl:app-locales-changed");
@@ -125,7 +218,16 @@ var SidebarUI = {
     let enumerator = Services.wm.getEnumerator("navigator:browser");
     if (!enumerator.hasMoreElements()) {
       let xulStore = Services.xulStore;
-      xulStore.persist(this._box, "sidebarcommand");
+
+      if (!this._keepTabsActivePref) {
+        xulStore.persist(this._box, "sidebarcommand");
+      }
+
+      // Remove the sidebar seperator with class sidebar-divider
+      const sidebarDivider = document.querySelector(".sidebar-divider");
+      if (sidebarDivider) {
+        sidebarDivider.remove();
+      }
 
       if (this._box.hasAttribute("positionend")) {
         xulStore.persist(this._box, "positionend");
@@ -157,7 +259,7 @@ var SidebarUI = {
   /**
    * The handler for Services.obs.addObserver.
    **/
-  observe(_subject, topic, _data) {
+  observe(_subject, topic, data) {
     switch (topic) {
       case "intl:app-locales-changed": {
         if (this.isOpen) {
@@ -168,6 +270,54 @@ var SidebarUI = {
           this.showInitially(this.lastOpenedId);
           break;
         }
+        break;
+      }
+
+      case "nsPref:changed": {
+        switch (data) {
+          case this.SIDEBAR_TABS_PREF:
+            this.setSidebarVisibility(
+              Services.prefs.getBoolPref(this.SIDEBAR_TABS_PREF, true)
+            );
+            break;
+
+          case this.SIDEBAR_KEEP_TABS_ACTIVE_PREF:
+            //Hide the sidebar if it is open
+            if (this.isOpen) {
+              this.hide();
+            }
+
+            this.uninit();
+
+            // We need to clean up all existing sidebar frames to prevent a memory leak.
+            for (const loadedSidebar of this.loadedSidebars) {
+              const elementId = `sidebar-browser-${loadedSidebar}`;
+              const element = document.getElementById(elementId);
+
+              if (element) {
+                element.remove();
+              }
+            }
+
+            // The sidebar depends on a browse existing. If this has been
+            // destroyed, we need to re-init it. Although, first we need to
+            // remove any possible cached value
+            this._browser = null;
+            if (!this.browser) {
+              /** @type {HTMLElement} */
+              const browser = document.createXULElement("browser");
+              browser.setAttribute("id", "sidebar");
+              browser.setAttribute("autoscroll", "false");
+              browser.setAttribute("disablehistory", "true");
+              browser.setAttribute("disablefullscreen", "true");
+              browser.setAttribute("tooltip", "aHTMLTooltip");
+              this._box.appendChild(browser);
+            }
+
+            this.init();
+            break;
+        }
+        break;
       }
     }
   },
@@ -284,17 +434,32 @@ var SidebarUI = {
     // First reset all ordinals to match DOM ordering.
     let browser = document.getElementById("browser");
     [...browser.children].forEach((node, i) => {
+      // Do not change ordinal on vertical tabs
+      if (node.id == "TabsToolbar" || node.id == "verticaltabs-splitter") {
+        return;
+      }
+
       node.style.MozBoxOrdinalGroup = i + 1;
     });
 
     if (!this._positionStart) {
-      // DOM ordering is:     |  sidebar-box  | splitter |   appcontent  |
-      // Want to display as:  |   appcontent  | splitter |  sidebar-box  |
-      // So we just swap box and appcontent ordering
+      // DOM ordering is:     | sidebar-container |  sidebar-box  | splitter |   appcontent  |
+      // Want to display as:  |   appcontent  | splitter |  sidebar-box | sidebar-container |
+      // So we just swap
+      // - sidebar-container and appcontent
+      // - splitter and sidebar-box
+
+      // sidebar-container and appcontent
       let appcontent = document.getElementById("appcontent");
-      let boxOrdinal = this._box.style.MozBoxOrdinalGroup;
-      this._box.style.MozBoxOrdinalGroup = appcontent.style.MozBoxOrdinalGroup;
+      let boxOrdinal = this._sidebarTabs.style.MozBoxOrdinalGroup;
+      this._sidebarTabs.style.MozBoxOrdinalGroup =
+        appcontent.style.MozBoxOrdinalGroup;
       appcontent.style.MozBoxOrdinalGroup = boxOrdinal;
+
+      // splitter and sidebar-box
+      const splitterBoxOrdinal = this._splitter.style.MozBoxOrdinalGroup;
+      this._splitter.style.MozBoxOrdinalGroup = this._box.style.MozBoxOrdinalGroup;
+      this._box.style.MozBoxOrdinalGroup = splitterBoxOrdinal;
       // Indicate we've switched ordering to the box
       this._box.setAttribute("positionend", true);
     } else {
@@ -483,6 +648,10 @@ var SidebarUI = {
   },
 
   _loadSidebarExtension(commandID) {
+    if (this._keepTabsActivePref && this.loadedSidebars.includes(commandID)) {
+      return;
+    }
+
     let sidebar = this.sidebars.get(commandID);
     let { extensionId } = sidebar;
     if (extensionId) {
@@ -521,6 +690,7 @@ var SidebarUI = {
       }
 
       this._fireFocusedEvent();
+      this.loadedSidebars.push(commandID);
       return true;
     });
   },
@@ -544,6 +714,7 @@ var SidebarUI = {
     }
     return this._show(commandID).then(() => {
       this._loadSidebarExtension(commandID);
+      this.loadedSidebars.push(commandID);
       return true;
     });
   },
@@ -557,6 +728,38 @@ var SidebarUI = {
    */
   _show(commandID) {
     return new Promise(resolve => {
+      if (this._keepTabsActivePref) {
+        const browserID = `sidebar-browser-${commandID}`;
+
+        // Check if the browser element for this command has already been
+        // created.
+        if (!document.getElementById(browserID)) {
+          // If it doesn't exist, we should create it. Note that this
+          // function will set this._browser to be this new browser
+          this.createBrowser(browserID, commandID);
+        } else {
+          // We need to set the active browser to be the one that already exists
+          this._browser = document.getElementById(browserID);
+        }
+
+        // We must ensure ensure that all non-active browsers are hidden.
+        let sidebarBrowsers = document.querySelectorAll(
+          "[id^='sidebar-browser-']"
+        );
+
+        for (const browser of sidebarBrowsers) {
+          // If the browser matches the one we want to display, make sure it is
+          // not hidden
+          if (browser.id == browserID) {
+            browser.removeAttribute("hidden");
+            continue;
+          }
+
+          // Otherwise, hide all other browsers
+          browser.setAttribute("hidden", "true");
+        }
+      }
+
       this.selectMenuItem(commandID);
 
       this._box.hidden = this._splitter.hidden = false;
@@ -574,7 +777,13 @@ var SidebarUI = {
       this.observeTitleChanges(sourceL10nEl);
       this.browser.setAttribute("src", url); // kick off async load
 
-      if (this.browser.contentDocument.location.href != url) {
+      const activeSidebarUnloaded =
+        this._keepTabsActivePref && !this.loadedSidebars.includes(commandID);
+      const regularSidebarUnloaded =
+        !this._keepTabsActivePref &&
+        this.browser.contentDocument.location.href != url;
+
+      if (activeSidebarUnloaded || regularSidebarUnloaded) {
         this.browser.addEventListener(
           "load",
           event => {
@@ -618,12 +827,19 @@ var SidebarUI = {
     // create a new content viewer because the old one doesn't get destroyed
     // until about:blank has loaded (which does not happen as long as the
     // element is hidden).
-    this.browser.setAttribute("src", "about:blank");
-    this.browser.docShell.createAboutBlankContentViewer(null, null);
+    if (!this._keepTabsActivePref) {
+      this.browser.setAttribute("src", "about:blank");
+      this.browser.docShell.createAboutBlankContentViewer(null, null);
+    }
 
     this._box.removeAttribute("checked");
     this._box.hidden = this._splitter.hidden = true;
+    this._sidebarIcons.removeAttribute("checked");
 
+    // Note: This was was commented out but had no description for its removed
+    // reason. I am bringing it back unless there is any other justification - tricky
+    //
+    // Removed in: :mute: Remove Test Logs
     let selBrowser = gBrowser.selectedBrowser;
     selBrowser.focus();
     if (triggerNode) {
@@ -636,10 +852,27 @@ var SidebarUI = {
    * none if the argument is an empty string.
    */
   selectMenuItem(commandID) {
+    // Pulse modification: This needs to update both which sidebar item is
+    // checked and which dropdown item is checked. Hence, we are overriding this
+    // method.
+
     for (let [id, { menuId, triggerButtonId }] of this.sidebars) {
+      if (!menuId) {
+        continue;
+      }
+
       let menu = document.getElementById(menuId);
-      let triggerbutton =
+      let dropdownTriggerButton =
         triggerButtonId && document.getElementById(triggerButtonId);
+      const sidebarTriggerButton = document.getElementById(
+        `sidebar-background-${id}`
+      );
+
+      setCheckedForItem(id, menu, dropdownTriggerButton);
+      setCheckedForItem(id, menu, sidebarTriggerButton);
+    }
+
+    function setCheckedForItem(id, menu, triggerbutton) {
       if (id == commandID) {
         menu.setAttribute("checked", "true");
         if (triggerbutton) {
@@ -655,6 +888,107 @@ var SidebarUI = {
       }
     }
   },
+
+  // ===========================================================================
+  // Pulse methods
+
+  /**
+   * Create sidebar browser between active and inactive for persistence reasons
+   *
+   * @todo What the hell does this comment mean?
+   */
+  createBrowser(id, commandID) {
+    let browser = document.createXULElement("browser");
+    browser.setAttribute("id", id);
+    browser.setAttribute("autoscroll", "false");
+    browser.setAttribute("disablehistory", "true");
+    browser.setAttribute("disablefullscreen", "true");
+    browser.setAttribute("tooltip", "aHTMLTooltip");
+    browser.setAttribute("class", "sidebar-tab");
+    if (commandID) {
+      browser.setAttribute("sidebarcommand", commandID);
+    }
+    browser.setAttribute("style", "min-width: 14em; -moz-box-flex: 1;");
+    this._box.appendChild(browser);
+    this._browser = browser;
+  },
+
+  /**
+   * Set the sidebar tabs to be visible or not visible
+   * @param {boolean} visible Sets the sidebar to be visible or not visible
+   */
+  setSidebarVisibility(visible) {
+    const sidebarContainer = this._sidebarTabs;
+
+    if (visible) {
+      sidebarContainer.style.display = "flex";
+    } else {
+      sidebarContainer.style.display = "none";
+    }
+  },
+
+  async createSidebarItem(id, config, isinit = false) {
+    if (!isinit) {
+      await this.promiseDeferred;
+    }
+
+    // if the sidebar item is already created, do nothing
+    if (document.getElementById(`sidebar-background-${id}`)) {
+      document.getElementById(`sidebar-background-${id}`).remove();
+    }
+
+    //if id contains ext
+    if (id.includes("ext")) {
+      if (!document.getElementsByClassName("sidebar-divider").length) {
+        const divider = document.createElement("hr");
+        divider.classList.add("sidebar-divider");
+        this._sidebarIcons.appendChild(divider);
+      }
+    }
+
+    const background = document.createXULElement("vbox");
+    background.classList.add("sidebar-item-background");
+    background.setAttribute("id", `sidebar-background-${id}`);
+
+    if (config.url) {
+      background.onclick = () => {
+        this.toggle(id);
+      };
+    }
+
+    if (config.onclick) {
+      background.setAttribute("onclick", config.onclick);
+    }
+
+    const iconEl = document.createElement("image");
+    iconEl.setAttribute("id", `sidebar-icon-${id}`);
+    iconEl.setAttribute("src", `${config.iconurl}`);
+    iconEl.style.backgroundImage = `url(${config.iconurl})`;
+
+    background.append(iconEl);
+
+    //If the item is meant to be on bottom, add it to the bottom
+    if (config.isBottom) {
+      this._sidebarIconsBottom.appendChild(background);
+    } else {
+      this._sidebarIcons.appendChild(background);
+    }
+
+    setTimeout(() => {
+      iconEl.setAttribute("class", "sidebar-icon-item");
+    }, 100);
+  },
+
+  deleteSidebarItem(id) {
+    const background = document.getElementById(`sidebar-background-${id}`);
+    const icon = document.getElementById(`sidebar-icon-${id}`);
+    if (background) {
+      background.remove();
+    }
+    if (icon) {
+      icon.remove();
+    }
+  },
 };
 
 // Add getters related to the position here, since we will want them
