diff --git a/browser/base/content/browser-box.inc.xhtml b/browser/base/content/browser-box.inc.xhtml
index 2c8edad944b52e2c8752cbeb55b8a1c62f01a961..16898ef8c3d1ef797a8fed17e9e8ce50992c364f 100644
--- a/browser/base/content/browser-box.inc.xhtml
+++ b/browser/base/content/browser-box.inc.xhtml
@@ -3,6 +3,11 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 <hbox flex="1" id="browser">
+  <div id="sidebar-container" class="sidebar-container"> 
+    <vbox id="sidebar-icons"></vbox>
+    <vbox id="sidebar-icons-bottom"></vbox>
+  </div>
+
   <vbox id="sidebar-box" hidden="true" class="chromeclass-extrachrome">
     <box id="sidebar-header" align="center">
       <toolbarbutton id="sidebar-switcher-target" flex="1" class="tabbable">
