diff --git a/browser/locales/en-US/browser/preferences/preferences.ftl b/browser/locales/en-US/browser/preferences/preferences.ftl
index b10403800c7aafe97471474f6bddb6098ce21a82..b39593816149a71a1e6b40626e304e367359fd02 100644
--- a/browser/locales/en-US/browser/preferences/preferences.ftl
+++ b/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -1413,3 +1413,11 @@ httpsonly-radio-disabled =
 desktop-folder-name = Desktop
 downloads-folder-name = Downloads
 choose-download-folder-title = Choose Download Folder:
+
+## Pulse browser: Settings entries localization
+preferences-sidebar-header = Sidebar
+preferences-sidebar-visible =
+    .label = Show sidebar tabs 
+
+preferences-sidebar-extensions-visible =
+    .label = Show available extensions on the sidebar
