diff --git a/browser/locales/en-US/browser/preferences/preferences.ftl b/browser/locales/en-US/browser/preferences/preferences.ftl
index 498b6aaf4bc60d3caa4d6a90423fd3dfb1c48b0b..bef1ecadb886bdda03217405537e8a5b66256a8f 100644
--- a/browser/locales/en-US/browser/preferences/preferences.ftl
+++ b/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -1364,3 +1364,11 @@ choose-download-folder-title = Choose Download Folder:
 #   $service-name (String) - Name of a cloud storage provider like Dropbox, Google Drive, etc...
 save-files-to-cloud-storage =
     .label = Save files to { $service-name }
+
+## Pulse browser: Settings enteries localization
+preferences-sidebar-header = Sidebar
+preferences-sidebar-visible =
+    .label = Show sidebar tabs 
+
+preferences-sidebar-extensions-visible =
+    .label = Show available extensions on the sidebar
