diff --git a/browser/locales/en-US/browser/preferences/preferences.ftl b/browser/locales/en-US/browser/preferences/preferences.ftl
index d7b87eacb425cb277e226261c7d994e084100a5d..7b8b924fa5a3456bc1c36b8f6fe1e9f378630782 100644
--- a/browser/locales/en-US/browser/preferences/preferences.ftl
+++ b/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -1424,3 +1424,12 @@ httpsonly-radio-disabled =
 desktop-folder-name = Desktop
 downloads-folder-name = Downloads
 choose-download-folder-title = Choose Download Folder:
+
+## Pulse browser: Settings entries localization
+preferences-sidebar-header = Sidebar
+preferences-sidebar-visible =
+    .label = Show sidebar tabs
+
+preferences-vertical-tabs-header = Vertical Tabs
+preferences-vertical-tabs-enabled =
+    .label = Use vertical tabs
