diff --git a/browser/themes/BuiltInThemeConfig.sys.mjs b/browser/themes/BuiltInThemeConfig.sys.mjs
index 9c93c0d39ad11d95ecb4ad119d3c21da2528af23..e00efac497130363a41c0ecdbe07df761076b368 100644
--- a/browser/themes/BuiltInThemeConfig.sys.mjs
+++ b/browser/themes/BuiltInThemeConfig.sys.mjs
@@ -32,6 +32,20 @@ export const BuiltInThemeConfig = new Map([
       path: "resource://builtin-themes/dark/",
     },
   ],
+  [
+    "pulse-light@browser.fushra.com",
+    {
+      version: "1.2",
+      path: "resource://builtin-themes/pulse/light/",
+    },
+  ],
+  [
+    "pulse-dark@browser.fushra.com",
+    {
+      version: "1.2",
+      path: "resource://builtin-themes/pulse/dark/",
+    },
+  ],
   [
     "firefox-alpenglow@mozilla.org",
     {
