diff --git a/browser/themes/ThemeVariableMap.sys.mjs b/browser/themes/ThemeVariableMap.sys.mjs
index ff8e85336158e4d86fab047882d6430b9e781445..4da903c46d8597ab1f749c9464ebcc0fd2f99c92 100644
--- a/browser/themes/ThemeVariableMap.sys.mjs
+++ b/browser/themes/ThemeVariableMap.sys.mjs
@@ -159,6 +159,18 @@ export const ThemeVariableMap = [
       },
     },
   ],
+  [
+    "--in-content-link-color",
+    {
+      lwtProperty: "icon_color",
+    },
+  ],
+  [
+    "--in-content-link-color-hover",
+    {
+      lwtProperty: "icon_color",
+    },
+  ],
 ];
 
 export const ThemeContentPropertyList = [
@@ -170,4 +182,10 @@ export const ThemeContentPropertyList = [
   "sidebar_highlight_text",
   "sidebar_text",
   "zap_gradient",
+
+  // Pulse
+  "button_primary",
+  "button_primary_hover",
+  "button_primary_active",
+  "button_primary_color",
 ];
