diff --git a/browser/themes/ThemeVariableMap.sys.mjs b/browser/themes/ThemeVariableMap.sys.mjs
index f2fd99193ddd0c24b7ad3ec953216973b162aaed..1810528d3b1d7b5cd2fb3f45548c8377a2f5f165 100644
--- a/browser/themes/ThemeVariableMap.sys.mjs
+++ b/browser/themes/ThemeVariableMap.sys.mjs
@@ -126,7 +126,9 @@ export const ThemeVariableMap = [
     "--sidebar-background-color",
     {
       lwtProperty: "sidebar",
-      optionalElementID: "sidebar-box",
+      // Pulse: we want to use the sidebar color in the sidebar tabs, which are
+      // under the `browser` parent rather than the `sidebar-box` parent
+      optionalElementID: "browser",
       processColor(rgbaChannels, element) {
         if (!rgbaChannels) {
           element.removeAttribute("lwt-sidebar");
@@ -187,4 +189,10 @@ export const ThemeContentPropertyList = [
   "sidebar_highlight_text",
   "sidebar_text",
   "zap_gradient",
+
+  // Pulse
+  "button_primary",
+  "button_primary_hover",
+  "button_primary_active",
+  "button_primary_color",
 ];
