diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index ee03d959d0222cbcd7f2b4b3a133edc483e03979..acd8615a3c2d0482a5eddc187c33137bf716067e 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -372,6 +372,16 @@ class UrlbarInput {
         !dueToSessionRestore &&
         (!this.window.isBlankPageURL(uri.spec) ||
           uri.schemeIs("moz-extension"));
+    
+      // Check if the addon is tabliss (our new tab) and stop if from showing up
+      // as such
+      try {
+        const { WebExtensionPolicy } = Cu.getGlobalForObject(Services);
+        const policy = WebExtensionPolicy.getByURI(uri);
+        if (policy.id === 'extension@tabliss.io') {
+          valid = false
+        }
+      } catch (e) {}     
     } else if (
       this.window.isInitialPage(value) &&
       lazy.BrowserUIUtils.checkEmptyPageOrigin(
