diff --git a/browser/components/extensions/jar.mn b/browser/components/extensions/jar.mn
index 12136c4d31cb8d4b7ba89139a8cfae52672c1b28..353146f52652daae6589f32a999e3d52de963fe5 100644
--- a/browser/components/extensions/jar.mn
+++ b/browser/components/extensions/jar.mn
@@ -14,6 +14,7 @@ browser.jar:
 #ifdef XP_WIN
     content/browser/extension-win-panel.css
 #endif
+    content/browser/parent/ext-sidebars.js (parent/ext-sidebars.js)
     content/browser/ext-browser.json
     content/browser/parent/ext-bookmarks.js (parent/ext-bookmarks.js)
     content/browser/parent/ext-browser.js (parent/ext-browser.js)
