diff --git a/toolkit/modules/moz.build b/toolkit/modules/moz.build
index c0c6e76515978b05ae9ed0281f0106469df2b573..12f3a1820f79430f5baeaf7ad51dcf61c1bb0c29 100644
--- a/toolkit/modules/moz.build
+++ b/toolkit/modules/moz.build
@@ -160,6 +160,7 @@ EXTRA_JS_MODULES += [
     "Color.sys.mjs",
     "Console.jsm",
     "ContentDOMReference.jsm",
+    "ContextMenu.jsm",
     "CreditCard.sys.mjs",
     "DateTimePickerPanel.sys.mjs",
     "DeferredTask.jsm",
