diff --git a/browser/themes/addons/jar.mn b/browser/themes/addons/jar.mn
index 271106757c7f60cdf78ff89c2dff375b524d197c..999049782628d501296a76990484fa29a5c3d5a9 100644
--- a/browser/themes/addons/jar.mn
+++ b/browser/themes/addons/jar.mn
@@ -16,6 +16,14 @@ browser.jar:
   content/builtin-themes/light                     (light/*.css)
   content/builtin-themes/light/manifest.json       (light/manifest.json)
 
+  content/builtin-themes/pulse/dark                      (pulse/dark/*.svg)
+  content/builtin-themes/pulse/dark/experiment.css       (pulse/dark/experiment.css)
+  content/builtin-themes/pulse/dark/manifest.json        (pulse/dark/manifest.json)
+
+  content/builtin-themes/pulse/light                     (pulse/light/*.svg)
+  content/builtin-themes/pulse/light/experiment.css      (pulse/light/experiment.css)
+  content/builtin-themes/pulse/light/manifest.json       (pulse/light/manifest.json)
+
   content/builtin-themes/monochromatic/lush/soft                          (monochromatic/lush/soft/*.svg)
   content/builtin-themes/monochromatic/lush/soft/manifest.json            (monochromatic/lush/soft/manifest.json)
   content/builtin-themes/monochromatic/lush/balanced                      (monochromatic/lush/balanced/*.svg)
