diff --git a/browser/branding/branding-common.mozbuild b/browser/branding/branding-common.mozbuild
index 908553b8b95c6997885f0c37baabbdf48e08614a..9f03c6fe53f56aa5fb36989044361d526e72708f 100644
--- a/browser/branding/branding-common.mozbuild
+++ b/browser/branding/branding-common.mozbuild
@@ -15,6 +15,8 @@ def FirefoxBranding():
             'pref/firefox-branding.js',
         ]
 
+    JS_PREFERENCE_FILES += ['pref/better-fox.js']
+
     if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'windows':
         FINAL_TARGET_FILES['..'] += [
             'firefox.VisualElementsManifest.xml',
