diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index 99aa63874bc6f1821baae6788fbba18f6861f9c6..76354d828578fabd297c817e15b7492c167958b9 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -14,6 +14,7 @@ var { ExtensionError } = ExtensionUtils;
 
 var { IconDetails } = ExtensionParent;
 
+
 // WeakMap[Extension -> SidebarAction]
 let sidebarActionMap = new WeakMap();
 
@@ -95,6 +96,9 @@ this.sidebarAction = class extends ExtensionAPI {
 
     this.tabContext.shutdown();
 
+    // Delete sidebar item
+    SidebarUI.deleteSidebarItem(this.id);
+
     // Don't remove everything on app shutdown so session restore can handle
     // restoring open sidebars.
     if (isAppShutdown) {
@@ -129,6 +133,8 @@ this.sidebarAction = class extends ExtensionAPI {
       if (SidebarUI.lastOpenedId === sidebarId) {
         SidebarUI.lastOpenedId = null;
       }
+      // Remove the sidebar item from the sidebar menu.
+      SidebarUI.deleteSidebarItem(sidebarId);
     }
   }
 
@@ -157,21 +163,30 @@ this.sidebarAction = class extends ExtensionAPI {
     }
     let { document, SidebarUI } = window;
     let keyId = `ext-key-id-${this.id}`;
+    let getIcon = size =>
+      IconDetails.escapeUrl(
+        IconDetails.getPreferredIcon(details.icon, this.extension, size).icon
+      );
 
     SidebarUI.sidebars.set(this.id, {
       title: details.title,
       url: sidebarURL,
       menuId: this.menuId,
+      iconurl: getIcon(16),
       buttonId: this.buttonId,
       // The following properties are specific to extensions
       extensionId: this.extension.id,
       panel: details.panel,
       browserStyle: this.browserStyle,
     });
+    
+    SidebarUI.createSidebarItem(this.id, SidebarUI.sidebars.get(this.id));
 
     let header = document.getElementById("sidebar-switcher-target");
     header.addEventListener("SidebarShown", this.updateHeader);
 
+
+
     // Insert a menuitem for View->Show Sidebars.
     let menuitem = document.createXULElement("menuitem");
     menuitem.setAttribute("id", this.menuId);
