diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 494ddaeb4ef50993a70230ed005ccb95b2684144..b7572b7f4c7b2fc5791d235141b4fd8db4178919 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -271,6 +271,7 @@
 ; All the pref files must be part of base to prevent migration bugs
 @RESPATH@/browser/@PREF_DIR@/firefox.js
 @RESPATH@/browser/@PREF_DIR@/firefox-branding.js
+@RESPATH@/browser/@PREF_DIR@/better-fox.js
 @RESPATH@/greprefs.js
 @RESPATH@/defaults/autoconfig/prefcalls.js
 @RESPATH@/browser/defaults/permissions
