diff --git a/devtools/client/performance-new/index.xhtml b/devtools/client/performance-new/index.xhtml
index e77548dcbd98a598822e080805d0e320310f3103..3011e0c0b84f9002b499f1ac2343c6c7bd34b372 100644
--- a/devtools/client/performance-new/index.xhtml
+++ b/devtools/client/performance-new/index.xhtml
@@ -13,6 +13,7 @@
   <head>
     <link rel="stylesheet" href="chrome://devtools/skin/widgets.css" type="text/css"/>
     <link rel="stylesheet" href="chrome://devtools/skin/perf.css" type="text/css"/>
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
   <body class="theme-body">
     <div id="root"></div>
