diff --git a/python/mozboot/mozboot/osx.py b/python/mozboot/mozboot/osx.py
index 8c8c478d2ef6bb7a0ba45a67cfb694d13fce3716..82ed833ea14db8de47e5aec9dd0f049e99b5973c 100644
--- a/python/mozboot/mozboot/osx.py
+++ b/python/mozboot/mozboot/osx.py
@@ -284,16 +284,23 @@ class OSXBootstrapper(OSXAndroidBootstrapper, BaseBootstrapper):
             self.install_homebrew()
 
     def ensure_clang_static_analysis_package(self):
-        from mozboot import static_analysis
+        # NOTE: This function has been removed as it caused numerous problems
+        # with the boostrap process for v100.0.2. May be fixed in the future,
+        # so we can return it
+        pass
+        # from mozboot import static_analysis
 
-        self.install_toolchain_static_analysis(static_analysis.MACOS_CLANG_TIDY)
+        # self.install_toolchain_static_analysis(static_analysis.MACOS_CLANG_TIDY)
 
     def ensure_sccache_packages(self):
         from mozboot import sccache
 
         self.install_toolchain_artifact("sccache")
         self.install_toolchain_artifact(sccache.RUSTC_DIST_TOOLCHAIN, no_unpack=True)
-        self.install_toolchain_artifact(sccache.CLANG_DIST_TOOLCHAIN, no_unpack=True)
+        # NOTE: This function has been removed as it caused numerous problems
+        # with the boostrap process for v100.0.2. May be fixed in the future,
+        # so we can return it
+        # self.install_toolchain_artifact(sccache.CLANG_DIST_TOOLCHAIN, no_unpack=True)
 
     def ensure_fix_stacks_packages(self):
         self.install_toolchain_artifact("fix-stacks")
