diff --git a/browser/components/customizableui/CustomizableUI.jsm b/browser/components/customizableui/CustomizableUI.jsm
index 87aede1449d7ac4f30e13990f6d928fba3484162..32deb776daa5f4a353667e282f2ebfd904cefedb 100644
--- a/browser/components/customizableui/CustomizableUI.jsm
+++ b/browser/components/customizableui/CustomizableUI.jsm
@@ -13,6 +13,10 @@ const { AppConstants } = ChromeUtils.import(
   "resource://gre/modules/AppConstants.jsm"
 );
 
+const { makeWidgetId } = ChromeUtils.import(
+  "resource://gre/modules/ExtensionCommon.jsm"
+).ExtensionCommon;
+
 XPCOMUtils.defineLazyModuleGetters(this, {
   AddonManager: "resource://gre/modules/AddonManager.jsm",
   AddonManagerPrivate: "resource://gre/modules/AddonManager.jsm",
@@ -32,6 +36,7 @@ XPCOMUtils.defineLazyGetter(this, "gWidgetsBundle", function() {
 });
 
 const kDefaultThemeID = "default-theme@mozilla.org";
+const kUBlockOriginID = "uBlock0@raymondhill.net";
 
 const kSpecialWidgetPfx = "customizableui-special-";
 
@@ -232,15 +237,15 @@ var CustomizableUIInternal = {
 
     let navbarPlacements = [
       "back-button",
-      "forward-button",
       "stop-reload-button",
+      "forward-button",
       Services.policies.isAllowed("removeHomeButtonByDefault")
         ? null
         : "home-button",
       "spring",
+      `${makeWidgetId(kUBlockOriginID)}-browser-action`,
       "urlbar-container",
       "spring",
-      "save-to-pocket-button",
       "downloads-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
       "fxa-toolbar-menu-button",
