diff --git a/browser/components/customizableui/CustomizableUI.jsm b/browser/components/customizableui/CustomizableUI.jsm
index e97c979363775d2de999e65e7e41c02a736f57ac..501f1235ac9ba571be66b533772cc259aceef48f 100644
--- a/browser/components/customizableui/CustomizableUI.jsm
+++ b/browser/components/customizableui/CustomizableUI.jsm
@@ -24,6 +24,7 @@ XPCOMUtils.defineLazyModuleGetters(lazy, {
   AddonManagerPrivate: "resource://gre/modules/AddonManager.jsm",
   SearchWidgetTracker: "resource:///modules/SearchWidgetTracker.jsm",
   CustomizableWidgets: "resource:///modules/CustomizableWidgets.jsm",
+  ExtensionCommon: "resource://gre/modules/ExtensionCommon.jsm",
   PanelMultiView: "resource:///modules/PanelMultiView.jsm",
   BrowserUsageTelemetry: "resource:///modules/BrowserUsageTelemetry.jsm",
   HomePage: "resource:///modules/HomePage.jsm",
@@ -36,6 +37,7 @@ XPCOMUtils.defineLazyGetter(lazy, "gWidgetsBundle", function() {
 });
 
 const kDefaultThemeID = "default-theme@mozilla.org";
+const kUBlockOriginID = "uBlock0@raymondhill.net";
 
 const kSpecialWidgetPfx = "customizableui-special-";
 
@@ -253,17 +255,19 @@ var CustomizableUIInternal = {
       false
     );
 
+    const { makeWidgetId } = lazy.ExtensionCommon;
+
     let navbarPlacements = [
       "back-button",
-      "forward-button",
       "stop-reload-button",
+      "forward-button",
       Services.policies.isAllowed("removeHomeButtonByDefault")
         ? null
         : "home-button",
       "spring",
+      `${makeWidgetId(kUBlockOriginID)}-browser-action`,
       "urlbar-container",
       "spring",
-      "save-to-pocket-button",
       "downloads-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
       "fxa-toolbar-menu-button",
V_EDITION ? "developer-button" : null,
       "fxa-toolbar-menu-button",
