diff --git a/browser/components/privatebrowsing/content/aboutPrivateBrowsing.js b/browser/components/privatebrowsing/content/aboutPrivateBrowsing.js
index c69aa3c1c1640217b5c1eba1457096230d3fcf9a..222af36287e7037a0505ca4c4ec5dab76467c222 100644
--- a/browser/components/privatebrowsing/content/aboutPrivateBrowsing.js
+++ b/browser/components/privatebrowsing/content/aboutPrivateBrowsing.js
@@ -167,7 +167,7 @@ async function renderPromo({
 }
 
 const DEFAULT_PRIVATE_BROWSING_CONTENT = {
-  promoEnabled: true,
+  promoEnabled: false,
   infoEnabled: true,
   infoIcon: "",
   infoTitle: "",
