diff --git a/browser/locales/en-US/browser/browser.ftl b/browser/locales/en-US/browser/browser.ftl
index de82989814fa2c4c08e58230dbdfac4e3ed61801..64243a0212516e2fe2ecf29d363ac22a4e84bf7c 100644
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -981,6 +981,9 @@ firefox-relay-offer-what-relay-provides = All emails sent to your email masks wi
 
 firefox-relay-offer-legal-notice = By clicking “Use email mask”, you agree to the <label data-l10n-name="tos-url">Terms of Service</label> and <label data-l10n-name="privacy-url">Privacy Notice</label>.
 
+sidebar-context-delete = 
+    .label = Delete
+
 ## Add-on Pop-up Notifications
 
 popup-notification-addon-install-unsigned =
