diff --git a/browser/locales/en-US/browser/browser.ftl b/browser/locales/en-US/browser/browser.ftl
index ca4bfcb81b852966a4486d8ec238345f18971f85..738fbe237412aacb4f24c84986c97bac0cf674b4 100644
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -950,3 +950,8 @@ refresh-blocked-redirect-label = { -brand-short-name } prevented this page from
 refresh-blocked-allow =
     .label = Allow
     .accesskey = A
+
+
+sidebar-context-delete = 
+    .label = Delete
+
