diff --git a/browser/locales/en-US/browser/browser.ftl b/browser/locales/en-US/browser/browser.ftl
index 710e5248557993012830e41d025b530c7bf33ade..8f58c6b9a2e6d726a30f49b40839b2c70e8d7195 100644
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -982,6 +982,9 @@ firefox-relay-offer-how-we-integrate = By continuing, you’ll be able to genera
 #  $useremail (String): user email that will receive messages
 firefox-relay-offer-what-relay-does = We’ll forward all emails from <strong>{ $sitename }</strong> to <strong>{ $useremail }</strong>.
 
+sidebar-context-delete = 
+    .label = Delete
+
 ## Popup Notification
 
 popup-notification-xpinstall-prompt-learn-more = Learn more about installing add-ons safely
